src/Simulation/QsharpCore/Core.qs

```diff
@@ -30,6 +30,21 @@ namespace Microsoft.Quantum.Core {
     @Attribute()
     newtype Deprecated = (NewName : String);
 
+    /// # Summary
+    /// Compiler-recognized attribute used to mark a callable with the runtime capabilities it requires.
+    ///
+    /// # Input
+    /// ## Level
+    /// The name of the runtime capability level required by the callable.
+    /// 
+    /// # Remarks
+    /// The valid capability level names, in order of increasing capabilities, are:
+    ///
+    /// 1. QPRGen0
+    /// 2. QPRGen1
+    /// 3. Unknown
+    @Attribute()
+    newtype Capability = (Level : String);
```

Comment:
Would it make sense to include in the name of this attribute that this isn't a capability itself, but a marker that a given capability is required? E.g.: `RequiredCapability`?

Reply:
Yeah, that is probably a better name. I think I would slightly prefer `RequiresCapability`.

Reply:
That kind of points to attributes-as-adjectives more generally, yeah.